sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs

```diff
@@ -171,11 +183,24 @@ private int CompareBodies(byte[] requestBody, byte[] responseBody, StringBuilder
             IsEquivalentUri(entry.RequestUri, otherEntry.RequestUri) &&
             CompareHeaderDictionaries(entry.Request.Headers, otherEntry.Request.Headers, VolatileHeaders) == 0;
 
-        private static bool AreUrisSame(string entryUri, string otherEntryUri) =>
-            // Some versions of .NET behave differently when calling new Uri("...")
-            // so we'll normalize the recordings (which may have been against
-            // a different .NET version) to be safe
-            new Uri(entryUri).ToString() == new Uri(otherEntryUri).ToString();
+        private bool AreUrisSame(string entryUri, string otherEntryUri) =>
+            TranslateUri(entryUri) == TranslateUri(otherEntryUri);
+
+        private string TranslateUri(string uriToTranslate)
```

Comment:
Nit - are you translating or normalizing/canonicalizing?

Reply:
I'm not sure any of these fit perfectly, but I guess Normalize is better.